org.eclipse.scout.rt.oauth2/src/main/java/org/eclipse/scout/rt/oauth2/OAuth2Helper.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2010-2022 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.oauth2;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.exception.ProcessingException;
+import org.eclipse.scout.rt.platform.util.date.DateUtility;
+import org.eclipse.scout.rt.platform.util.date.IDateProvider;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+@ApplicationScoped
+public class OAuth2Helper {
+
+  private Lock m_cacheAccessLock = new ReentrantLock();
+  private Map<String, TokenEntry> m_tokenCache = new HashMap<>();
+
+  /**
+   * Callers of this method must ensure, that oAuth2Config.getId() gives a unique String for each OAuth2 configuration
```

Comment:
Remove ID and work with full bean as key (implement equals/hashcode)? If full configuration is relevant, it's easier that way and doesn't require to work with a synthetic ID. Seems a bit strange here.